sdk/internal/recording/recording.go

```diff
@@ -433,3 +438,221 @@ var modeMap = map[RecordMode]recorder.Mode{
 	Live:     recorder.ModeDisabled,
 	Playback: recorder.ModeReplaying,
 }
+
+var recordMode, _ = os.LookupEnv("AZURE_RECORD_MODE")
+
+const (
+	modeRecording      = "record"
+	modePlayback       = "playback"
+	baseProxyURLSecure = "localhost:5001"
+	baseProxyURL       = "localhost:5000"
+	IdHeader           = "x-recording-id"
+	ModeHeader         = "x-recording-mode"
+	UpstreamUriHeader  = "x-recording-upstream-base-uri"
+)
+
+var recordingId string
+
+var tr = &http.Transport{
+	TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+}
+var client = http.Client{
+	Transport: tr,
+}
+
+type RecordingOptions struct {
+	UseHTTPS bool
+	Host     string
+	Scheme   string
+}
+
+func resetRecordingId() {
+	recordingId = ""
+}
+
+func defaultOptions() *RecordingOptions {
+	return &RecordingOptions{
+		UseHTTPS: true,
+		Host:     "localhost:5001",
+		Scheme:   "https",
+	}
+}
+
+func (r RecordingOptions) HostScheme() string {
+	if r.UseHTTPS {
+		return "https://localhost:5001"
+	}
+	return "http://localhost:5000"
+}
+
+func getTestId(pathToRecordings string, t *testing.T) string {
+	return pathToRecordings + "/recordings/" + t.Name() + ".json"
+}
+
+func StartRecording(t *testing.T, pathToRecordings string, options *RecordingOptions) error {
+	if options == nil {
+		options = defaultOptions()
+	}
+	if recordMode == "" {
+		t.Log("AZURE_RECORD_MODE was not set, options are \"record\" or \"playback\". \nDefaulting to playback")
+		recordMode = "playback"
+	}
+	testId := getTestId(pathToRecordings, t)
```

Comment:
Should probably validate that 'recordMode' is a valid value (ie, not just empty, but actually spelled correctly, etc..)